File: BrokeClientTS/src/clientPoolSwimmer.ts
import {ClientPool} from "./clientPool";
import {Query} from "./common/query";
import {SwimmerResponse} from "./models/SwimmerResponse";

export class ClientPoolSwimmer {
    private clientPool: ClientPool;
    public Id: string;
    constructor(clientPool: ClientPool, a: SwimmerResponse) {
        this.clientPool = clientPool;
        this.Id = a.Id;
    }
    public SendMessage(query: Query): void {
        query.Add("~ToSwimmer~", this.Id);
        this.clientPool.clientBrokerManager.client.SendMessage(query);
    }
    public SendMessageWithResponse<T>(query: Query, callback: (response: T) => void): void {
        query.Add("~ToSwimmer~", this.Id);
        this.clientPool.clientBrokerManager.client.SendMessageWithResponse(query, (response) => {
            callback(response.GetJson<T>());
        });
    }
}

File: BrokerClient/ClientPoolSwimmer.cs
using System;
using BrokerCommon;
using BrokerCommon.Models;

namespace BrokerClient
{
    public class ClientPoolSwimmer
    {
        private ClientPool clientPool;
        public string Id { get; set; }

        public ClientPoolSwimmer(ClientPool clientPool, SwimmerResponse a)
        {
            this.clientPool = clientPool;
            this.Id = a.Id;
        }


        public void SendMessage(Query query)
        {
            query.Add("~ToSwimmer~", this.Id);
            clientPool.clientBrokerManager.client.SendMessage(query);
        }

        public void SendMessageWithResponse<T>(Query query, Action<T> callback) where T : class
        {
            query.Add("~ToSwimmer~", this.Id);
            clientPool.clientBrokerManager.client.SendMessageWithResponse(query, (response) =>
            {
                callback(response.GetJson<T>());
            });
        }
    }
}